walk/walk.go

```diff
@@ -137,13 +138,13 @@ func Walk(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode,
 //
 // Traversal may skip subtrees or files based on the config.Config exclude/ignore/follow options
 // as well as the UpdateFilter callbacks.
-func visit(c *config.Config, cexts []config.Configurer, knownDirectives map[string]bool, updateRels *UpdateFilter, trie *pathTrie, wf WalkFunc, rel string, updateParent bool) ([]string, bool) {
+func visit(c *config.Config, cexts []config.Configurer, knownDirectives map[string]bool, updateRels *UpdateFilter, trie *pathTrie, wf WalkFunc, updateParent bool) ([]string, bool) {
 	haveError := false
 
 	// Absolute path to the directory being visited
-	dir := filepath.Join(c.RepoRoot, rel)
+	dir := filepath.Join(c.RepoRoot, trie.rel)
 
-	f, err := loadBuildFile(c, rel, dir, trie.files)
+	f, err := trie.build, trie.buildFileErr
 	if err != nil {
```

Comment:
This error logging moving to `buildTrie` would remove the need for the `trie.buildFileError` field and (imo) clean everything up, but that would be a minor change in behaviour that we can discuss later.

That would introduce the assumption that `c.Strict` is global (which it is atm) and not modifiable per-BUILD like other `c.*` fields :/